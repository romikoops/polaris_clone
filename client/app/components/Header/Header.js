import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavDropdown } from '../NavDropdown/NavDropdown';
import styles from './Header.scss';
import accountIcon from '../../assets/images/icons/person-dark.svg';
import defs from '../../styles/default_classes.scss';
class Header extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        const { user, theme } = this.props;
        const dropDownText = user && user.data ? user.data.first_name + ' ' + user.data.last_name : '';
        const dropDownImage = accountIcon;
        const accountLinks = [
            {
                url: '/account',
                text: 'Settings',
                fontAwesomeIcon: 'fa-cog',
                key: 'settings'
            },
            {
                url: '/signout',
                text: 'Sign out',
                fontAwesomeIcon: 'fa-sign-out',
                key: 'signOut'
            }
        ];
        const dropDown = user ? <NavDropdown
                                dropDownText={dropDownText}
                                dropDownImage={dropDownImage}
                                linkOptions={accountLinks}
                            /> : '';

        return (
            <div
                className={`${
                    styles.header
                } layout-row flex-100 layout-wrap layout-align-center`}
            >
                <div className={`${defs.content_width} layout-row flex-none`}>
                    <div className="layout-row flex-50 layout-align-start-center">
                        <img
                            src={theme ? theme.logoLarge : ''}
                            className={styles.logo}
                            alt=""
                        />
                    </div>
                    <div className="layout-row flex-50 layout-align-end-center">
                        {dropDown}
                    </div>
                </div>
            </div>
        );
    }
}

Header.propTypes = {
    tenant: PropTypes.object,
    theme: PropTypes.object,
    user: PropTypes.object,
    loggedIn: PropTypes.bool,
    shipment: PropTypes.object,
    dispatch: PropTypes.func,
    history: PropTypes.object,
    match: PropTypes.object
};

function mapStateToProps(state) {
    const { authentication, tenant, shipment } = state;
    const { user, loggedIn } = authentication;
    return {
        user,
        tenant,
        loggedIn,
        shipment
    };
}

export default connect(mapStateToProps)(Header);
