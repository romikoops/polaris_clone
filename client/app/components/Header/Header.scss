:local {
  .header {
    height: 100px;
    background-color: #ffffff;

    h4 {
      color: #000000;
    }
  }
}
