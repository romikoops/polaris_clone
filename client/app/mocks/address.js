export const firstAddress = {
  user: {
    primary: true
  },
  primary: true,
  id: 246,
  streetNumber: 'FOO_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_ADDRESS_STREET',
  city: 'FOO_ADDRESS_CITY',
  zipCode: 'FOO_ADDRESS_ZIP_CODE',
  country: 'FOO_ADDRESS_COUNTRY'
}

export const secondAddress = {
  user: {
    primary: false
  },
  primary: false,
  id: 704,
  streetNumber: 'BAR_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'BAR_GEOCODED_ADDRESS',
  street: 'BAR_ADDRESS_STREET',
  city: 'BAR_ADDRESS_CITY',
  zipCode: 'BAR_ADDRESS_ZIP_CODE',
  country: 'BAR_ADDRESS_COUNTRY'
}

export const originAddress = {
  city: 'ORIGIN_ADDRESS_CITY',
  country: 'ORIGIN_ADDRESS_COUNTRY',
  street: 'ORIGIN_ADDRESS_STREET',
  street_number: 'ORIGIN_ADDRESS_STREET_NUMBER',
  zip_code: 'ORIGIN_ADDRESS_ZIP_CODE'
}
export const destinationAddress = {
  city: 'DESTINATION_ADDRESS_CITY',
  country: 'DESTINATION_ADDRESS_COUNTRY',
  street: 'DESTINATION_ADDRESS_STREET',
  street_number: 'DESTINATION_ADDRESS_STREET_NUMBER',
  zip_code: 'DESTINATION_ADDRESS_ZIP_CODE'
}

export const addresses = {
  origin: originAddress,
  destination: destinationAddress
}
