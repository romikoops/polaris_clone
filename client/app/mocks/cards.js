export const firstCard = {
  name: 'FOO_NAME',
  img: 'FOO_IMG',
  url: 'FOO_URL',
  handleClick: x => x,
  options: {
    name: 'FOO_OPTIONS_NAME'
  }
}
export const secondCard = {
  name: 'BAR_NAME',
  img: 'BAR_IMG',
  url: 'BAR_URL',
  handleClick: x => x,
  options: {
    name: 'BAR_OPTIONS_NAME'
  }
}
export const cards = [
  firstCard,
  secondCard
]
