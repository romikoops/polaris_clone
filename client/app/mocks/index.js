export * from './address'
export * from './cards'
export * from './cargoItems'
export * from './client'
export * from './consignee'
export * from './contacts'
export * from './containers'
export * from './currencies'
export * from './dimensions'
export * from './documents'
export * from './feeHash'
export * from './functions'
export * from './gMaps'
export * from './group'
export * from './hub'
export * from './match'
export * from './notifyees'
export * from './offer'
export * from './results'
export * from './routes'
export * from './schedules'
export * from './scope'
export * from './shipmentData'
export * from './shipments'
export * from './shipper'
export * from './tenant'
export * from './theme'
export * from './user'
