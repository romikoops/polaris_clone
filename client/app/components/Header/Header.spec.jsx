import * as React from 'react'
import { shallow } from 'enzyme'
import { req, identity, tenant, theme, user } from '../../mocks'

jest.mock('react-redux', () => ({
  connect: (x, y) => Component => Component
}))
// eslint-disable-next-line
import Header from './Header'

const propsBase = {
  tenant,
  theme,
  user,
  registering: false,
  loggingIn: false,
  invert: false,
  loginAttempt: false,
  messageDispatch: {
    getUserConversations: identity
  },
  messages: [],
  showRegistration: false,
  unread: 0,
  req,
  scrollable: false,
  appDispatch: identity,
  noMessages: false,
  component: <div>FooComponent</div>
}

test('shallow render', () => {
  expect(shallow(<Header {...propsBase} />)).toMatchSnapshot()
})

test('noMessages is true', () => {
  const props = {
    ...propsBase,
    noMessages: true
  }
  expect(shallow(<Header {...props} />)).toMatchSnapshot()
})

test('unread > 0', () => {
  const props = {
    ...propsBase,
    unread: 1
  }
  expect(shallow(<Header {...props} />)).toMatchSnapshot()
})

test('user.role_id === 2', () => {
  const props = {
    ...propsBase,
    user: {
      ...user,
      role_id: 2
    }
  }
  expect(shallow(<Header {...props} />)).toMatchSnapshot()
})

test('user is falsy', () => {
  const props = {
    ...propsBase,
    user: null
  }
  expect(shallow(<Header {...props} />)).toMatchSnapshot()
})

test('showRegistration is true', () => {
  const props = {
    ...propsBase,
    showRegistration: true
  }
  expect(shallow(<Header {...props} />)).toMatchSnapshot()
})
