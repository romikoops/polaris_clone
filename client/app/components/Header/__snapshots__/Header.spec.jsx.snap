// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error && error[currentStage] 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="flex-none layout-row error_messages"
  >
    <FlashMessages
      messages={
        Array [
          Object {},
        ]
      }
    />
  </div>
</div>
`;

exports[`invert is true 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center inverted  "
  style={
    Object {
      "color": "white",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={true}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`noMessages is true 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`scrollable is true 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center  scrollable "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`shallow render 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`showModal is true 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
      <Modal
        classNames="ccb_login_modal"
        component={
          <LoginRegistrationWrapper
            LoginPageProps={
              Object {
                "req": null,
                "tenant": Object {
                  "emails": Object {
                    "sales": Object {
                      "air": "sales@demo.com",
                      "general": "sales@demo.com",
                      "ocean": "sales@demo.com",
                      "rail": "sales@demo.com",
                    },
                    "support": Object {
                      "air": "imc.air@demo.com",
                      "general": "support@demo.com",
                      "ocean": "imc.sea@demo.com",
                      "rail": "imc.rail@demo.com",
                    },
                  },
                  "id": 123,
                  "name": "TENANT_NAME",
                  "phones": Object {
                    "support": "TENANT_SUPPORT_PHONE",
                  },
                  "scope": Object {
                    "cargo_info_level": "text",
                    "carriage_options": Object {
                      "on_carriage": Object {
                        "export": "mandatory",
                        "import": "mandatory",
                      },
                      "pre_carriage": Object {
                        "export": "optional",
                        "import": "optional",
                      },
                    },
                    "closed_quotation_tool": true,
                    "customs_export_paper": false,
                    "detailed_billing": true,
                    "has_customs": true,
                    "has_insurance": true,
                    "hide_grand_total": true,
                    "mandatory_form_fields": Object {
                      "description_of_goods": "MANDATORY_FIELDS_DESCRIPTION",
                      "total_goods_value": 3370,
                    },
                    "modes_of_transport": Object {
                      "air": Object {
                        "AIR_LOAD_TYPE": 0,
                      },
                      "ocean": Object {
                        "OCEAN_LOAD_TYPE": 2,
                      },
                      "rail": Object {
                        "RAIL_LOAD_TYPE": 0,
                      },
                      "truck": Object {
                        "TRUCK_LOAD_TYPE": 0,
                      },
                    },
                    "terms": Array [
                      "FOO_TERM",
                      "BAR_TERM",
                    ],
                  },
                  "subdomain": "TENANT_SUBDOMAIN",
                  "theme": Object {
                    "colors": Object {
                      "primary": "#333",
                      "secondary": "#fafafa",
                    },
                  },
                },
                "theme": Object {
                  "colors": Object {
                    "primary": "#333",
                    "secondary": "#fafafa",
                  },
                },
              }
            }
            RegistrationPageProps={
              Object {
                "req": null,
                "tenant": Object {
                  "emails": Object {
                    "sales": Object {
                      "air": "sales@demo.com",
                      "general": "sales@demo.com",
                      "ocean": "sales@demo.com",
                      "rail": "sales@demo.com",
                    },
                    "support": Object {
                      "air": "imc.air@demo.com",
                      "general": "support@demo.com",
                      "ocean": "imc.sea@demo.com",
                      "rail": "imc.rail@demo.com",
                    },
                  },
                  "id": 123,
                  "name": "TENANT_NAME",
                  "phones": Object {
                    "support": "TENANT_SUPPORT_PHONE",
                  },
                  "scope": Object {
                    "cargo_info_level": "text",
                    "carriage_options": Object {
                      "on_carriage": Object {
                        "export": "mandatory",
                        "import": "mandatory",
                      },
                      "pre_carriage": Object {
                        "export": "optional",
                        "import": "optional",
                      },
                    },
                    "closed_quotation_tool": true,
                    "customs_export_paper": false,
                    "detailed_billing": true,
                    "has_customs": true,
                    "has_insurance": true,
                    "hide_grand_total": true,
                    "mandatory_form_fields": Object {
                      "description_of_goods": "MANDATORY_FIELDS_DESCRIPTION",
                      "total_goods_value": 3370,
                    },
                    "modes_of_transport": Object {
                      "air": Object {
                        "AIR_LOAD_TYPE": 0,
                      },
                      "ocean": Object {
                        "OCEAN_LOAD_TYPE": 2,
                      },
                      "rail": Object {
                        "RAIL_LOAD_TYPE": 0,
                      },
                      "truck": Object {
                        "TRUCK_LOAD_TYPE": 0,
                      },
                    },
                    "terms": Array [
                      "FOO_TERM",
                      "BAR_TERM",
                    ],
                  },
                  "subdomain": "TENANT_SUBDOMAIN",
                  "theme": Object {
                    "colors": Object {
                      "primary": "#333",
                      "secondary": "#fafafa",
                    },
                  },
                },
                "theme": Object {
                  "colors": Object {
                    "primary": "#333",
                    "secondary": "#fafafa",
                  },
                },
                "user": Object {
                  "company_name": "FOO_COMPANY",
                  "currency": "EUR",
                  "email": "foo@bar.baz",
                  "first_name": "John",
                  "guest": false,
                  "last_name": "Doe",
                  "phone": "6345789",
                  "role": Object {
                    "name": "shipper",
                  },
                  "role_id": 1,
                },
              }
            }
            initialCompName="LoginPage"
            t={[Function]}
            updateDimensions={null}
          />
        }
        flexOptions=""
        horizontalPadding="40px"
        maxWidth="90%"
        minHeight=""
        parentToggle={[Function]}
        showExit={false}
        verticalPadding="30px"
      />
    </div>
  </div>
</div>
`;

exports[`showRegistration is true 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`tenant is falsy 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`theme has logoLarge 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="logo"
        onClick={[Function]}
        src="LOGO_LARGE"
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`theme has logoWide 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="wide_logo"
        onClick={[Function]}
        src="LOGO_WIDE"
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`unread > 0 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`user && user.role && user.role.name.includes(admin) 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/admin/settings",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "admin",
              },
              "role_id": 1,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`user is falsy 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText=""
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={null}
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`user.role_id === 2 1`] = `
<div
  className="header layout-row flex-100 layout-wrap layout-align-center   "
  style={
    Object {
      "color": "black",
    }
  }
>
  <div
    className="flex-100 layout-row"
    style={
      Object {
        "padding": "0 15px",
      }
    }
  >
    <div
      className="hide-sm hide-xs layout-row flex layout-align-start-center pointy"
    >
      <img
        alt=""
        className="undefined"
        onClick={[Function]}
        src=""
        style={
          Object {
            "display": "block",
          }
        }
      />
    </div>
    <div>
      FooComponent
    </div>
    <div
      className="flex layout-row layout-align-end-center"
    >
      <div
        className="layout-row layout-align-space-around-center"
      >
        <NavDropdown
          dropDownText="John Doe"
          invert={false}
          isLanding={false}
          linkOptions={
            Array [
              Object {
                "fontAwesomeIcon": "fa-cog",
                "key": "settings",
                "text": "Account",
                "url": "/account/profile",
              },
              Object {
                "fontAwesomeIcon": "fa-sign-out",
                "key": "signOut",
                "select": [Function],
                "text": "Sign out",
              },
            ]
          }
          loginText="Log In / Register"
          toggleShowLogin={[Function]}
          user={
            Object {
              "company_name": "FOO_COMPANY",
              "currency": "EUR",
              "email": "foo@bar.baz",
              "first_name": "John",
              "guest": false,
              "last_name": "Doe",
              "phone": "6345789",
              "role": Object {
                "name": "shipper",
              },
              "role_id": 2,
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
